ipam/allocator.go

```diff
@@ -376,26 +376,31 @@ func (alloc *Allocator) Shutdown() {
 	<-doneChan
 }
 
-// AdminTakeoverRanges (Sync) - take over the ranges owned by a given peer.
+// AdminTakeoverRanges (Sync) - take over the ranges owned by a given
+// peer, and return how much space was transferred in the process.
 // Only done on adminstrator command.
-func (alloc *Allocator) AdminTakeoverRanges(peerNameOrNickname string) error {
-	resultChan := make(chan error)
+func (alloc *Allocator) AdminTakeoverRanges(peerNameOrNickname string) int {
+	resultChan := make(chan int)
 	alloc.actionChan <- func() {
 		peername, err := alloc.lookupPeername(peerNameOrNickname)
 		if err != nil {
-			resultChan <- fmt.Errorf("Cannot find peer '%s'", peerNameOrNickname)
+			alloc.warnf("attempt to take over range from unknown peer '%s'", peerNameOrNickname)
+			resultChan <- 0
 			return
 		}
 
 		alloc.debugln("AdminTakeoverRanges:", peername)
 		if peername == alloc.ourName {
-			resultChan <- fmt.Errorf("Cannot take over ranges from yourself!")
+			alloc.warnf("attempt to take over range from ourself")
+			resultChan <- 0
 			return
 		}
 
-		newRanges, err := alloc.ring.Transfer(peername, alloc.ourName)
-		alloc.space.AddRanges(newRanges)
-		resultChan <- err
+		before := alloc.space.NumFreeAddressesInRange(alloc.universe)
+		alloc.space.AddRanges(alloc.ring.Transfer(peername, alloc.ourName))
+		after := alloc.space.NumFreeAddressesInRange(alloc.universe)
+
+		resultChan <- int(after) - int(before)
```

Comment:
I hate that cast here. I cannot fathom why `NumFreeAddressesInRange` returns an `address.Offset`. The returned value does not fit any definition of 'offset' that I can think of. @bboreham @tomwilkie 

I suppose I could stick with that pattern and return an `address.Offset` from this function. Though I would still need a `Sub` or `Difference` function for offsets, returning offsets. Which is just weird.

Or perhaps leave the return type as is and have a function `Difference(address.Offset, address,Offset) int`.